WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1349,6 +1415,70 @@ private boolean handleBackPressed() {
         return true;
     }
 
+    private RemotePreviewLogicHelper.RemotePreviewHelperFunctions getEditPostActivityStrategyFunctions() {
+       return new RemotePreviewLogicHelper.RemotePreviewHelperFunctions() {
+            @Override
+            public boolean isNewPost() {
+                return EditPostActivity.this.isNewPost();
+            }
+
+            @Override
+            public void notifyNoNetwork() {
+                UploadUtils.showSnackbar(findViewById(R.id.editor_activity), R.string.no_network_message);
+            }
+
+            @Override
+            public boolean notifyUploadInProgress(PostModel post) {
+                if (UploadService.hasInProgressMediaUploadsForPost(post)) {
+                    ToastUtils.showToast(EditPostActivity.this,
+                            getString(R.string.editor_toast_uploading_please_wait), Duration.SHORT);
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+
+            @Nullable
+            @Override
+            public PostModel updatePostIfNeeded() {
+                updatePostObject();
+                return mPost;
+            }
+
+            @Override
+            public boolean canDiscard() {
+                return isDiscardable();
```

Comment:
I think we want to use `PostUtils.isPublishable` instead of `isDiscardable` here. I'd also rename it if we change that. I guess we can remove `canDiscard` from the helper completely (and use the PostUtils directly).

`isDiscardable` check if the post is "freshly created" but that should not impact what kind of save we do before showing the preview.

Reply:
Yep, removed canDiscard from the helper and used PostUtilityWrapper for testability/mocking purposes. Please see again e8e97532281aa2adce1528bce409a30f303dff00.